CMakePresets.json

```diff
@@ -60,7 +60,7 @@
           "generator": "Ninja",
           "cacheVariables": {
               "VCPKG_TARGET_TRIPLET": "$env{VCPKG_TARGET_TRIPLET}",
-              "VCPKG_ROOT": "$env{VCPKG_ROOT}",
+              "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
```

Comment:
Same deal here as with `remill` and `anvill`. If we want the presets to use the Clang that comes bundled with the LLVM install that we're building against, the presets need `VCPKG_ROOT`.

Reply:
I'm not sure I follow why the cache variable would be needed in addition to the env var?

Reply:
A few lines below, the preset sets the C and C++ compilers to the ones in the LLVM install (so something like `${VCPKG_ROOT}/installed/$env{VCPKG_TARGET_TRIPLET}/tools/llvm/clang`). If we remove `VCPKG_ROOT` as a cache variable and only have `CMAKE_TOOLCHAIN_FILE`, I'm not sure how we can do that.

Reply:
`"CMAKE_C_COMPILER": "$env{VCPKG_ROOT}/installed/$env{VCPKG_TARGET_TRIPLET}/tools/llvm/clang"` uses `$env{VCPKG_ROOT}`, not the cache var, or am I missing something?

Reply:
Are you suggesting that the `CMAKE_TOOLCHAIN_FILE` cache variable could be removed because it isn't used to find the compilers?

Reply:
`CMAKE_TOOLCHAIN_FILE` is used by Rellic to find the packages so it can't be removed. What confuses me is

> If we want the presets to use the Clang that comes bundled with the LLVM install that we're building against, the presets need `VCPKG_ROOT`.

I guess I don't understand which `VCPKG_ROOT` you're referring to, the env var or cache var?

Reply:
> I guess I don't understand which `VCPKG_ROOT` you're referring to, the env var or cache var?

I'm referring to the env var. I've removed all references of `VCPKG_ROOT`, but this is one spot where I'm not sure that I can.

Reply:
Ohhh sorry then 🤣 Keeping the `VCPKG_ROOT` is fine to me, the user needs to be able to specify where to find vcpkg in any case

Reply:
I am confused, but probably out of ignorance because I don't know anything about these presets files. Why is `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` needed in the presets?

Reply:
I don't think it's needed, it's just convenient. The presets are a handful of commonly used configurations and one way that they try to be helpful is by setting `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` to the Clang versions that belong to the LLVM install you're building against.

This gives you a relatively recent compiler which is probably helpful on systems where the default `cc` and `c++` are outdated. On the other hand, I think this is counterproductive for debug builds because you probably don't want to be building with a debug build of Clang.

I'd be happy to just get rid of this unless anyone else is opposed to it.

Reply:
If removing that doesn't break anything else, I'm all for it

Reply:
Ditto.